libraries/standard/mqtt/include/types/iot_mqtt_types.h

```diff
@@ -445,6 +445,7 @@ typedef struct IotMqttCallbackParam
      */
     IotMqttConnection_t mqttConnection;
 
+    /* coverity[misra_c_2012_rule_19_2_violation] */
```

Comment:
We should have a comment justifying this suppression, something about how we are using unions here to reduce struct size

Reply:
I didn't add that because I think that is a simple well-known general C knowledge, not specific to our code.  If you insist, I can add.

Reply:
We should at least have something about unions here, as otherwise whoever is reading this code would have to be familiar with or look up MISRA 19.2. Since 19.2 specifically says unions should not be used, we should say why we are using them here.